mlflow/pytorch/_pytorch_autolog.py

```diff
@@ -0,0 +1,208 @@
+                if isinstance(callback, pl.callbacks.early_stopping.EarlyStopping):
+                    self._early_stop_check(callback)
+
+        def on_train_start(self, trainer, pl_module):
+            """
+            Logs Optimizer related metrics when the train begins
+
+            :param trainer: pytorch lightning trainer instance
+            :param pl_module: pytorch lightning base module
+            """
+            try_mlflow_log(mlflow.set_tag, "Mode", "training")
+            try_mlflow_log(mlflow.log_param, "epochs", trainer.max_epochs)
+
+            for callback in trainer.callbacks:
+                if isinstance(callback, pl.callbacks.early_stopping.EarlyStopping):
+                    self.early_stopping = True
+                    self._log_early_stop_params(callback)
+
+            if hasattr(trainer, "optimizers"):
+                for optimizer in trainer.optimizers:
+                    try_mlflow_log(mlflow.log_param, "optimizer_name", type(optimizer).__name__)
+
+                    if hasattr(optimizer, "defaults"):
+                        defaults_dict = optimizer.defaults
+                        if "lr" in defaults_dict:
+                            try_mlflow_log(mlflow.log_param, "learning_rate", defaults_dict["lr"])
+
+                        if "eps" in defaults_dict:
+                            try_mlflow_log(mlflow.log_param, "epsilon", defaults_dict["eps"])
+
+                        if "betas" in defaults_dict:
+                            try_mlflow_log(mlflow.log_param, "betas", defaults_dict["betas"])
+
+                        if "weight_decay" in defaults_dict:
+                            try_mlflow_log(
+                                mlflow.log_param, "weight_decay", defaults_dict["weight_decay"]
+                            )
+
+            summary = str(ModelSummary(pl_module, mode="full"))
+            tempdir = tempfile.mkdtemp()
+            try:
+                summary_file = os.path.join(tempdir, "model_summary.txt")
+                with open(summary_file, "w") as f:
+                    f.write(summary)
+
+                try_mlflow_log(mlflow.log_artifact, local_path=summary_file)
+            finally:
+                shutil.rmtree(tempdir)
+
+        def on_train_end(self, trainer, pl_module):
+            """
+            Logs the model checkpoint into mlflow - models folder on the training end
+
+            :param trainer: pytorch lightning trainer instance
+            :param pl_module: pytorch lightning base module
+            """
+
+            mlflow.pytorch.log_model(pytorch_model=trainer.model, artifact_path="model")
+
+            if self.early_stopping and trainer.checkpoint_callback.best_model_path:
+                try_mlflow_log(
+                    mlflow.log_artifact,
+                    local_path=trainer.checkpoint_callback.best_model_path,
+                    artifact_path="restored_model_checkpoint",
+                )
```

Comment:
Should we log the checkpoint in the `"model"` directory?

Reply:
checkpoints are saved as `.ckpt`. I guess it would be confusing to have both `pytorch` and `pytorch_lightning` model under same folder `model`.

Reply:
@harupy This will get confusing for user's. The Checkpoints reside in their own separate folder

Reply:
Makes. Let's just leave this as it is.